SO3_XL_ToolChange_fixedLocation

```diff
@@ -31,67 +49,67 @@
 %FEEDRATE = modal.feedrate
 %SPINDLE = modal.spindle
 %COOLANT = modal.coolant
-; Stop spindle
-M5
-; Absolute positioning
+
+M5   ; Stop spindle
+
 G90
-; Go to Safe Z
 G53 Z[SAFE_HEIGHT]
-; Go to tool probe X,Y
 G53 X[TOOL_PROBE_X] Y[TOOL_PROBE_Y]
-G53 Z[TOOL_PROBE_Z]
-; Wait until the planner queue is empty
+
 %wait
-; Pause the program before probing
+
+; Set up for probing
 M0
-; Probe toward workpiece with a maximum probe distance
-G91 ; Relative positioning
-G38.2 Z-[PROBE_DISTANCE] F[PROBE_FEEDRATE]
-G0 Z2 ;lift 2mm
-G38.2 Z-5 F45 ;Probe Z
-G90 ; Absolute positioning
-%ORIGINAL_TOOL = [posz] ;store current work position
-; A dwell time of one half second to make sure the planner queue is empty
-G4 P0.5
-
-; Retract from the touch plate for tool change
-G91 ; Relative positioning
-G0 Z[RETRACTION_DISTANCE]
+
+G53 Z[TOOL_PROBE_Z]
+G91
+G38.2 Z-[PROBE_DISTANCE] F[PROBE_FAST_FEEDRATE]
+G0 Z1
+G38.2 Z-2 F[PROBE_SLOW_FEEDRATE]
+
+G90
+%ORIGINAL_TOOL = posz
```

Comment:
Why do you think this had an effect?
I've used the macro without issue in X & Y.

Reply:
I truly do not know. I didn't dig deep enough to learn how to introspect these values, but before making this change, I found that the X & Y was consistently being set incorrectly (by a random few mm in each direction).

I originally thought it was a bug that my changes introduced, so I went back to using your exact code (except for changing the tool change location offsets), and then commented out the meat of the macro so that all it was doing was moving to the tool change location, pausing, and then moving back to the original location. I started uncommenting things portion by portion, and eventually narrowed it down to this line: when it was `%ORIGINAL_TOOL = [posz]`, it would return to a slightly incorrect location, but when it was `%ORIGINAL_TOOL = posz`, it was correct. Why that is? No idea 😅